src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -81,6 +89,69 @@ class MatchService(
     }
   }
 
+  private fun nameMatch(matchRequest: MatchRequest): BoolQueryBuilder? {
+    with(matchRequest) {
+      return QueryBuilders.boolQuery()
+          .mustWhenTrue({ activeSentence }, "currentDisposal", "1")
+          .must(QueryBuilders.boolQuery()
+              .should(QueryBuilders.boolQuery()
+                  .mustWhenPresent("surname", surname)
+                  .mustWhenPresent("firstName", firstName)
+                  .mustWhenPresent("dateOfBirth", dateOfBirth)
+              )
+              .should(QueryBuilders.boolQuery()
+                  .mustWhenPresent("offenderAliases.surname", surname)
+                  .mustWhenPresent("offenderAliases.firstName", firstName)
+                  .mustWhenPresent("offenderAliases.dateOfBirth", dateOfBirth)
+              )
+          )
+    }
+  }
+
+  private fun partialNameMatch(matchRequest: MatchRequest): BoolQueryBuilder? {
+    with(matchRequest) {
+      return QueryBuilders.boolQuery()
+          .mustWhenTrue({ activeSentence }, "currentDisposal", "1")
+          .mustWhenPresent("surname", surname)
+          .mustWhenPresent("dateOfBirth", dateOfBirth)
```

Comment:
Any reason this doesn't check the aliases too?  (like nameMatch above)

Reply:
My hypothesis is when we are checking so few bits of information that also checking alias will be too weak a match to be useful.  So in this case we are just checking surname with date of birth, but if they previously gave their name and date of birth falsely there is reasonable chance they are using another identity..

Where say below it is probably fine to to use the firstName alias "John, Jon, Johhny"

But these is all hypothesis based on the limited manual matching I have done so far.

The next stage will be to get the code live and see what false positives we get and false negatives(with out actually updating any data!)